modules/react/common/lib/utils/components.ts

```diff
@@ -177,40 +176,46 @@ export const createComponent = <T extends React.ElementType | undefined = undefi
    *   )
    * }
    */
-  Component: RefForwardingComponent<T, P>;
+  Component: RefForwardingComponent<E, P>;
   /**
    * Used in container components
    */
   subComponents?: SubComponents;
-}): (T extends undefined
+}): (E extends undefined
   ? Component<P>
   : ElementComponent<
-      // T is not `undefined` here, but Typescript thinks it could be, so we add another `undefined`
+      // E is not `undefined` here, but Typescript thinks it could be, so we add another `undefined`
       // check and cast to a `React.FC` to match a valid signature for `ElementComponent`.
       // `React.FC` was chosen as the simplest valid interface.
-      T extends undefined ? React.FC : T,
+      E extends undefined ? React.FC : E,
       P
     >) &
   SubComponents => {
-  const ReturnedComponent = React.forwardRef<T, P & {as?: React.ElementType}>(
+  const ReturnedComponent = React.forwardRef<E, P & {as?: React.ElementType}>(
     ({as: asOverride, ...props}, ref) => {
       return Component(
         props as P,
-        // @ts-ignore React considers no passed ref as `null`, Typescript considers this
```

Comment:
This `ts-ignore` is an internal type and won't be in the definition file, but I think we should prefer `any` over `ts-ignore` as a bail to avoid issues for users.